### DISCUSSION TITLE
An Infinite number of error messages appear after git push

### DISCUSSION THREAD

**Discussion by Phobooky** [xxyzz/WordDumb#59] [Q&A] [answered]
After instaling [worddumb-8a3f79e422791c3d20a66ad96007c187fc915369](https://github.com/xxyzz/WordDumb/suites/7657939258/artifacts/318959625), it appears this extrange behabiour.
Preferences->Customize ePub Dictionary->Select Spanish (in my case). An Infinite number of error messages appear. The only way is cancel Calibre.
Aspect (the shadow in the images is related with a lot of winwdows):
![image](https://user-images.githubusercontent.com/24461946/182800419-4e4ed996-bcd5-49ad-acdc-c1b02dc8292a.png)

**Comment by xxyzz** [comment, marked as answer]
I added IPA and POS column to the lemmas table in the previous commits, you'll need to delete the old `wiktionary_*` files in the `worddumb-lemmas` folder.

**Comment by Phobooky** [comment]
Just after solving the previous problem, now new issues arise.
All books produce error when generating x-ray and wordwise.
Example:

```
calibre, version 6.2.0 (win32, embedded-python: True)
Tonnerre de Brest!: An error occurred, please copy error message then report bug at GitHub.


Starting job: Generating Word Wise and X-Ray for The Kill Artist 
Job: "Generating Word Wise and X-Ray for The Kill Artist" failed with error: 
Traceback (most recent call last):
  File "calibre\gui2\threaded_jobs.py", line 82, in start_work
  File "calibre_plugins.worddumb.parse_job", line 154, in do_job
  File "calibre_plugins.worddumb.parse_job", line 305, in create_files
  File "calibre_plugins.worddumb.epub", line 191, in modify_epub
  File "calibre_plugins.worddumb.epub", line 310, in create_word_wise_footnotes
ValueError: not enough values to unpack (expected 4, got 3)
 
Called with args: ((161, 'EPUB', 'C:\\Archivos de Programas\\Calibre\\Calibre Library\\Daniel Silva\\The Kill Artist (161)\\The Kill Artist - Daniel Silva.epub', <calibre.ebooks.metadata.book.base.Metadata object at 0x00000195F6675060>, {'spacy': 'en_core_web_', 'wiki': 'en', 'kaikki': 'English'}), True, True) {'notifications': <queue.Queue object at 0x00000195F6676230>, 'abort': <threading.Event object at 0x00000195F66744F0>, 'log': <calibre.utils.logging.GUILog object at 0x00000195F6674A00>} 
```
Another one:

```calibre, version 6.2.0 (win32, embedded-python: True)
Tonnerre de Brest!: An error occurred, please copy error message then report bug at GitHub.

Starting job: Generating Word Wise and X-Ray for La última noche en Tremore Beach 
Job: "Generating Word Wise and X-Ray for La última noche en Tremore Beach" failed with error: 
Traceback (most recent call last):
  File "calibre\gui2\threaded_jobs.py", line 82, in start_work
  File "calibre_plugins.worddumb.parse_job", line 154, in do_job
  File "calibre_plugins.worddumb.parse_job", line 305, in create_files
  File "calibre_plugins.worddumb.epub", line 191, in modify_epub
  File "calibre_plugins.worddumb.epub", line 310, in create_word_wise_footnotes
ValueError: not enough values to unpack (expected 4, got 3)
 
Called with args: ((155, 'EPUB', 'C:\\Archivos de Programas\\Calibre\\Calibre Library\\Mikel Santiago\\La ultima noche en Tremore Beach (155)\\La ultima noche en Tremore Beac - Mikel Santiago.epub', <calibre.ebooks.metadata.book.base.Metadata object at 0x000001960B5306A0>, {'spacy': 'es_core_news_', 'wiki': 'es', 'kaikki': 'Spanish'}), True, True) {'notifications': <queue.Queue object at 0x000001960B530CA0>, 'abort': <threading.Event object at 0x000001960DFA3400>, 'log': <calibre.utils.logging.GUILog object at 0x000001960DFA3D30>} 
```

**Reply by xxyzz** [reply]
Maybe you didn't delete the old `wiktionary_es_dump` file?

**Comment by Phobooky** [comment]
Yes, I deleted it, but the new one generated is very small comparing with the previous one:

![image](https://user-images.githubusercontent.com/24461946/182811660-1543bd9d-a8a1-41ff-b510-0af18a4eedaa.png)

**Reply by xxyzz** [reply]
I deleted some words that have the "misspelling", "form-of" and "alt-of" tags.

**Comment by Phobooky** [comment]
And... I was supposed to do something with the errors generating files?

**Reply by xxyzz** [reply]
I'm not sure what you mean, did you encounter another error?

**Comment by Phobooky** [comment]
I mean.... you told me that you had deleted some words. But I don not what it does mean in reference to my problem. My conversions still goes wrong with the last git push.

**Reply by xxyzz** [reply]
I said I deleted some words to answer your question about the decreased Wiktionary file size. I though the `not enough values to unpack` error was solved once you deleted all the old `wiktioanry_*` files and created the new files. I tried to add Word Wise to a Spanish EPUB book and the job finished without an error.

**Reply by xxyzz** [reply]
There suppose to be a "wiktionary_es_tst" file in the folder created by the latest commit plugin, but your screenshot doesn't have it. It's only used for searching lemma but it's not there, kind strange.

**Comment by Phobooky** [comment]
I've check the complete installation and it already works perfectly now. But, I have a doubt. The wiktionary_es.json file size has decreased a lot. Do you think it is correct?

![image](https://user-images.githubusercontent.com/24461946/182875493-eb27ca02-ba0d-4b88-b4f7-4173c70611a9.png)

Another thing. I've observed that many activated words in previous version of "Customize EPUB Wiktionary", now are disabled (maked as false). For example, the word "ramshackle" was marked as true and now it is marked as false.  I can control the level of difficulty in Kindle Word Wise using the device. But when I generate an epub with wordwise using "Customize EPUB Wiktionary", as every word has not level of difficulty.... How can I manage the level of difficulty generanding WordWise in EPUBs?

**Reply by xxyzz** [reply]
I removed some inflected words from the Spanish Wiktionary in recent commits, that's why the file size decreased. 

"ramshackle": fixed in https://github.com/xxyzz/WordDumb/commit/763f1593adee89bdb5e0ebdd2eda06070373d280

Difficulty is only used on Kindle. Word Wise is added as footnote on EPUB, I don't think EPUB can filter footnotes.

**Comment by Phobooky** [comment]
Now I can see it is alright. Thanks.

**Comment by Phobooky** [comment]
But, related with the fact that difficulty is only used on Kindle, I can see the percentaje of words with Word Wise on EPUB is very diferent depending of the language as you can see in these two pages. I don't know if it would have a meaning. 

![image](https://user-images.githubusercontent.com/24461946/183036497-e773112c-b8d6-4e2b-a0ce-45e845f59787.png)

![image](https://user-images.githubusercontent.com/24461946/183036631-4e853141-3d91-4ad5-880f-d36f5d5c6430.png)

**Reply by xxyzz** [reply]
I only enabled English Wiktionary words that are enabled for Kindle. For other languages I enabled all of them because I don't have the similar prevalence data as English.

**Comment by Phobooky** [comment]
Ok. I understand. :)
